presto-jdbc/src/test/java/io/prestosql/jdbc/TestPrestoDriverUri.java

```diff
@@ -213,6 +214,60 @@ public void testUriWithSslEnabledPassword()
         assertEquals(properties.getProperty(SSL_TRUST_STORE_PASSWORD.getKey()), "password");
     }
 
+    @Test
+    public void testUriWithExtraCredentials()
+            throws SQLException
+    {
+        String extraCredentials = "test.token.foo:bar;test.token.abc:xyz";
+        PrestoDriverUri parameters = createDriverUri("presto://localhost:8080?extraCredentials=" + extraCredentials);
+        Properties properties = parameters.getProperties();
+        assertEquals(properties.getProperty(EXTRA_CREDENTIALS.getKey()), extraCredentials);
+    }
+
+    @Test(expectedExceptions = SQLException.class)
```

Comment:
Move all these checks for invalid URLs to `testInvalidUrls()` which will allow validating the error messages